File: api/ECommerce.Api/Models/Category.cs
using System.ComponentModel.DataAnnotations;

namespace ECommerce.Api.Models
{
    public class Category
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;

        [Required]
        [MaxLength(2048)]
        public string Image { get; set; } = string.Empty;

        public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAtUtc { get; set; }
        public bool IsActive { get; set; } = true;

        public int SortOrder { get; set; }
    }
}

File: sportapp/src/types/category.ts
export interface Category {
  id: string;
  name: string;
  description: string;
  image: string;
  isActive: boolean;
  sortOrder: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateCategoryRequest {
  name: string;
  description: string;
  image: string;
  isActive: boolean;
  sortOrder: number;
}

export interface UpdateCategoryRequest {
  name: string;
  description: string;
  image: string;
  isActive: boolean;
  sortOrder: number;
}
